src/ruby/puppet-server-lib/puppet/server/http_client.rb

```diff
@@ -70,12 +79,24 @@ def get(url, headers)
     request_options.set_as(ResponseBodyType::TEXT)
     configure_timeouts(request_options)
     configure_ssl(request_options)
-    response = SyncHttpClient.get(request_options)
+    response = client_get(request_options)
     ruby_response(response)
   end
 
   private
 
+  def client_get(request_options)
+    SyncHttpClient.get(request_options)
+  rescue Java::ComPuppetlabsHttpClient::HttpClientException => e
+    raise Puppet::Server::HttpClientError.new(e.inspect, e)
```

Comment:
The way I'd been thinking about it is that is that the first argument you give to the `HttpClientError` constructor becomes what is returned back from the `message` method on the object which is created so it might make most sense to just extract the `message` from the HttpClientException and pass that as-is into the `HttpClientError` that we're re-raising.  It just felt a little weird to me to take the output from `inspect` on one object and have that become `message` on a different object.

Practically speaking, I think the difference in doing that is only that the name of the Java Exception class will end up not getting appended to the `message` in the `HttpClientError` Ruby class.

@jeffmccune - do you have any recommendations / preferences here?